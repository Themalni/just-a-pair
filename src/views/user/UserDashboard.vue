<template>
  <div class="dashboard mt-6">
    <h1 class="mt-6 d-block">Welcome User, {{ name }}</h1>
    <button class="btn-default" @click="logout()">Logout</button>
  </div>
</template>

<script>
import { fb } from "@/firebase";

export default {
  name: "UserDashboard",
  data() {
    return {
      name: null,
      email: null
    };
  },
  created() {
    var user = fb.auth().currentUser;
    this.email = user.email;
    this.name = user.displayName;

    console.log("Profile from FB:", this.profile);
  },
  firestore() {
    const user = fb.auth().currentUser;
    return {
      profile: db.collection("profiles").doc(user.uid)
    };
  },
  methods: {
    logout() {
      fb.auth()
        .signOut()
        .then(() => {
          console.log("User has been signed out!");
          this.$router.push("/");
        });
    }
  }
};
</script>
<style lang="scss" scoped>
.dashboard {
  height: 300px;

  h1 {
    margin-top: 50px;
  }
}
</style>
