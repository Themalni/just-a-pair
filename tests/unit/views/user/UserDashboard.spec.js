import { mount, createLocalVue } from '@vue/test-utils';
import UserDashboard from "@/views/user/UserDashboard";
import mockState from '../../utils/mockState';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('UserDashboard', () => {
    let wrapper
    let store
    let state

    beforeEach(() => {
        state = mockState.state
        store = new Vuex.Store({
            state: state
        })

        wrapper = mount(UserDashboard, {
            stubs: ['md-icon'],
            store,
            localVue
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('renders correctrly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
})