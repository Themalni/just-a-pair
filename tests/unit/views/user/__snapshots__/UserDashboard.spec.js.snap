// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboard renders correctrly 1`] = `
<div class="dashboard mt-6 d-flex">
  <div class="sidebar-container">
    <h3 class="sidebar-header pl-3 pt-4 pb-3">Your profile</h3>
    <div class="separator"></div>
    <div class="sidebar-user mt-3 ml-3 mb-2">Hello, John Smith</div>
    <div class="sidebar-user mt-3 ml-3 mb-2">john_doe@gmail.com</div>
    <div class="separator mt-3 mb-4"></div> <button class="logout-btn button-reset text-white ml-3">
      <md-icon-stub class="icon-sm">logout</md-icon-stub>Logout
    </button>
  </div>
  <div class="content p-4">
    <h3>Profile details</h3>
    <p>You can change you profile details below</p>
    <form>
      <div class="form-group"><label for="user-name">Name</label> <input type="text" id="user-name" placeholder="Name" class="form-control"></div>
      <div class="form-group"><label for="user-email">E-mail</label> <input type="text" id="user-email" placeholder="E-mail" class="form-control"></div>
      <div class="form-group"><label for="user-password">Password</label> <input type="password" id="user-password" placeholder="Password" class="form-control"></div> <button type="submit" disabled="disabled" class="btn btn-sm btn-primary ml-0">Save</button>
    </form>
  </div>
</div>
`;
